cluster-autoscaler/core/scaleup/orchestrator/orchestrator.go

```diff
@@ -546,6 +588,15 @@ func (o *ScaleUpOrchestrator) IsNodeGroupResourceExceeded(resourcesLeft resource
 		klog.Errorf("Skipping node group %s; error getting node group resources: %v", nodeGroup.Id(), err)
 		return NotReadyReason
 	}
+	autoscalingOptions, aErr := nodeGroup.GetOptions(o.autoscalingContext.NodeGroupDefaults)
+	if aErr != nil {
+		klog.Errorf("Couldn't get autoscaling options for ng: %v", nodeGroup.Id())
+	}
+	if autoscalingOptions != nil && autoscalingOptions.AtomicScaleUp {
+		for resource, delta := range resourcesDelta {
+			resourcesDelta[resource] = delta * int64(nodeGroup.MaxSize())
```

Comment:
This should be maxsize - targetsize to work correctly. But even then, I think it would be better to introduce either new function or extending `IsNodeGroupResourceExceeded` to take number of nodes as an argument - putting this piece of logic here is confusing.

Reply:
Extended the `IsNodeGroupResourceExceeded` to take in number of nodes as well, and now it calculates resource delta with the given number of nodes.